### PR TITLE
Auto-magically generated section navigation

### PR DESCRIPTION
**Pull request by johnpuddephatt**

### CODE REVIEW

File: _includes/nav-docs.html [hunk 1]
```diff
@@ -1,21 +1,18 @@
...
-          </ul>
-        {% endif %}
-      </li>
-    {% endfor %}
-  </ul>
-{% endfor %}
+    {% assign pages = site.pages | sort:"weight"  %}
+    {% assign current_array = page.url | remove_first: "/" | split:"/" %}
+    {% assign current_section = current_array[0] %}
+        <h3>{{ current_section | replace: '-',' ' | capitalize }}</h3>
```
Comment (bernardodiasc) [inline _includes/nav-docs.html]: extra indentation here. also, please check your https://github.com/symphonycms/docs.getsymphony.com/blob/gh-pages/.editorconfig (this is not mandatory, but good enforce coding standards, which is 2 spaces for indentation)

Thanks!

File: _includes/nav-primary.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{% assign current_array = page.url | remove_first: "/" | split:"/" %}
+{% assign current_section = current_array[0] %}
+
+<nav>
+{% for section in site.data.nav_primary %}
```
Comment (bernardodiasc) [inline _includes/nav-primary.html]: check the indentation here.

File: search.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ---
+template: default
```
Comment (bernardodiasc) [inline search.json]: why we need this?

**Comment by bernardodiasc** [comment]
hey @johnpuddephatt, I made some notes on the PR. Also let's use on `.md` file extension, to keep consistent. :)

Thank you!

**Comment by johnpuddephatt** [comment]
Sorry, my bad on the formatting :disappointed: 

I've sorted most of this but have his a small snag: I want to set a default layout (so that it doesn't have to be set in each page file) but this applies the layout to javascript file too. I've tried setting the default to only apply to pages but this still causes the problem to occur.

I'll keep trying, hopefully find a fix shortly.

**Comment by bernardodiasc** [comment]
as far as I know, empty double `---` will make jekyll build the file but not apply any layout. this is used in the json and js file, the first because is buit with liquid, the second because I need the `site.baseurl` there.

**Comment by johnpuddephatt** [comment]
This is true, but not if a default layout value is set in `_config.yml`.  We don't have to do this but I think it makes sense as it means not having to set the layout on every page.

I found a fix which was to set the layout to a non-existent layout. The other option would be to create an empty layout file, but I'm not sure there would be any benefit to this.

Anyway, all working now. And hopefully the indentation is fixed, let me know if I've missed anything.

**Comment by johnpuddephatt** [comment]
The ability to only set a default by type is [documented here](http://jekyllrb.com/docs/configuration/#front-matter-defaults) btw, but I had no luck with it. It would have been my preferred solution. Despite putting `type: "pages"`, the js file still got dropped onto the default layout.

**Comment by bernardodiasc** [comment]
does `search.json` need `layout: none` too?

**Comment by johnpuddephatt** [comment]
Yep – I'm not sure how I missed that. Search is working again now :relieved:

**Comment by bernardodiasc** [comment]
Fantastic! Great job mate!

**Comment by johnpuddephatt** [comment]
:smile: I'll try and make sure future PR's are a bit smoother – I'm still getting the hang of Git/GitHub and collaborative working!

**Comment by bernardodiasc** [comment]
Way to go! It's good to have opportunity to learn by doing. :)

**Pull request merged**